pkg/server/plugin/nodeattestor/aws/session.go

```diff
@@ -35,22 +38,38 @@ func (cfg *SessionConfig) Validate(defaultAccessKeyID, defaultSecretAccessKey st
 }
 
 // newAWSSession create an AWS config from the credentials and given region
-func newAWSConfig(accessKeyID, secretAccessKey, region, assumeRoleArn string) aws.Config {
-	var credsProvider aws.CredentialsProvider
-	switch {
-	case assumeRoleArn != "":
-		stsConf := aws.Config{
-			Region: region,
+func newAWSConfig(accessKeyID, secretAccessKey, region, assumeRoleArn string) (aws.Config, error) {
+	ctx := context.Background()
+
+	var opts []func(*config.LoadOptions) error
```

Comment:
It's a little unfortunate that `config.LoadDefaultConfig()` accepts a `...func(*config.LoadOptions) error` rather than a `...config.LoadOptionsFunc`

Reply:
yeap